src/domain/alerts/contracts/abi-decoder.helper.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  Abi,
+  DecodeEventLogParameters,
+  DecodeFunctionDataParameters,
+  Hex,
+  decodeEventLog as _decodeEventLog,
+  decodeFunctionData as _decodeFunctionData,
+} from 'viem';
+
+export abstract class AbiDecoder<TAbi extends Abi | readonly unknown[]> {
+  readonly abi: TAbi;
+
+  constructor(abi: TAbi) {
+    this.abi = abi;
+  }
```

Comment:
We can also inline the property here and make it `private`. I don't believe it's being accessed externally.

```typescript
protected constructor(private readonly abi: TAbi) {}
```